# **Introduction to Statistics**

Stats is the science of counting, collecting, organizing and analyzing **data**

> - **Data:** "facts or pieces of information"

Examples:
1. Height of students in classroom.
2. No of sales of in terms of revenue of a company
3. Salary of people in society
4. IQ of student in classroom.

###  **Table of Contents**
1. Scale of Measurements
2. Measure of Central Tendency
3. Measure of Dispersion
4. Random Variable
5. Sets
6. Histogram and Skewness

## **1. Scale of Measurements**

There are two main categories under each two sub categories are there in which data is measured

- **Categorical**
    - Nominal Scale Data
    - Ordinal Scale Data

- **Numerical**
    - Interval Scale Data
    - Ratio Scale Data

**<u>1.1 Nominal Scale Data</u>**

- Qualitative/ Categorical data

- Eg., Gender, Color, Labels

- Order or Rank does `not` matter

Example: favorite color of girls.

| Color | Vote | Percentage |
|---|---|---|
| Red | 5 | 50% |
| Yellow | 3 | 30% |
| Orange | 2 | 20% |
|*Total*| *10* | *100%* |

**<u>1.2 Ordinal Scale Data</u>**

- Rank matters
- Order is important
- Difference cannot be mesured

Example: Top 3 runner ups of 100m race

| Ranks | Reach Time | Difference w.r.t 1<sup>st</sup>| Difference w.r.t 2<sup>nd</sup>|
|---|---|---|---|
| 1<sup>st</sup> | 12 Sec |  0 Sec | 0 Sec |
| 2<sup>nd</sup> | 15 Sec |  3 Sec | 0 Sec |
| 3<sup>rd</sup> | 20 Sec |  8 Sec | 5 Sec |

**<u>1.3 Interval Scale Data</u>**

- Here also order matters
- Difference can be measured
- The ratio cannot be measured
- No zero starting point

**Example 1 :** Temperature Variable in Fahrenheit (<sup>o</sup>F)

| Temperature | Difference w.r.t 1<sup>st</sup>| Difference w.r.t 2<sup>nd</sup>|
|---|---|---|
| 30<sup>o</sup>F | 0<sup>o</sup>F | 30<sup>o</sup>F |
| 60<sup>o</sup>F | 30<sup>o</sup>F | 0<sup>o</sup>F |
| 90<sup>o</sup>F | 60<sup>o</sup>F | 30<sup>o</sup>F |

- we can't measure ratio of temperature like : 60 <sup>o</sup>F / 30 <sup>o</sup>F == 2 : 1 
- we can't have **0 <sup>o</sup>F** or **0 <sup>o</sup>C** temperature in real world

if you did not got 1st example here is 2nd example.

**Example 2 :** IQ of students/persons
- IQ scores are designed to be a measure of cognitive abilities relative to the general population

| Person | IQ |
|---|---|
| 1 | 140 |
| 2 |  70 |
| 3 | 210 |

- IQ of a person `cannot` be `zero`
- IQ<sub>1</sub> / IQ<sub>2</sub> = 140 / 70 = 2:1
- from above , equation it says 1st person have 2x(twice) of the IQ of 2nd person
- but in reality, there is no such thing of having 2x cognitive abilities

**<u>1.4 Ratio Scale Data</u>**

- Order matters
- Differences are measurable (Ratio)
- Contain a zero starting point

**Example 1:** Students marks in exam.

students scores(out of 100): 0, 90, 60, 30, 75, 40, 50

| Student | Score |
|---|---|
| S1 | 0 |
| S2 |  90 |
| S3 | 60 |
| S4 | 30 |
| S5 | 75 |
| S6 | 40 |
| S7 | 50 |

- difference is scores
    - S2, S3 = 90 - 60 = 30 
- ratio in scores
    - S2 : S3 = 90 : 60 = 1.5 : 1 

## **2. Measure of Central Tendency**

there are 3 measures of central tendency as following

- Mean
- Median
- Mode

<img src="assets/measure_of_central_tendency.png" alt="Histograms" height="400">

### **1. Mean**

The mean provides a central value around which the data points are distributed

**Pupulation(N)**

`X = {1, 1, 2, 2, 3, 3, 4, 5, 5, 6}`

Population mean(&mu;) = $\mu = \frac{1}{N} \sum_{i=1}^{N} x_i$

```python

= (1 + 1 + 2 + 2 + 3 + 3 + 4 + 5 + 5 + 6)/10 = 32/10 = 3.2

```

In [3]:
X = [1,1,2,2,3,3,4,5,5,6]

sum_X = sum(X)

mu = sum_X / len(X)

print(mu)

3.2


**Sample(n)**

`X = {2, 3, 1, 4, 8, 9, 5}`

Sample mean(&mu;) = $\bar{x} = \frac{1}{n} \sum_{i=1}^{n} x_i$

```python

= (2 + 3 + 1 + 4 + 8 + 9 + 5) / 7 = 32/7 = 4.5

```

In [5]:
x = [2, 3, 1, 4, 8, 9, 5]

sum_x = sum(x)

x_bar = sum_x / len(x)

print(x_bar)

4.571428571428571


 let's check with extream high and low values, what happends to mean ?

`X = {2, 3, 1, 4, 8, 9, 5}`

In [31]:
# added -50
x = [-50, 2, 3, 1, 4, 8, 9, 5]

sum_x = sum(x)

x_bar = sum_x / len(x)

print(x_bar)

-2.25


- mean changed from `4.5` to `-2.25`
- moved to left, when we added low extream i.e -50

In [32]:
# added 50
x = [2, 3, 1, 4, 8, 9, 5, 50]

sum_x = sum(x)

x_bar = sum_x / len(x)

print(x_bar)

10.25


- mean changed from `4.5` to `10.25`
- moved to right, when we added high extream i.e 50

### **Properties**

- **Sensitive to Outliers:** The mean is affected by extremely high or low values, making it a sensitive measure of central tendency.

### **2. Median**

The median is a measure of central tendency that represents the middle value of a dataset when it is ordered from smallest to largest. It is especially useful for understanding the distribution of data, particularly when dealing with skewed distributions or outliers

**Formula and Calculation**

1. **Order the Data:** Arrange the data points in ascending order.

2. **Find the Median:**
    - *Odd Number of Values:* If there is an odd number of values, the median is the middle value.
    - *Even Number of Values:* If there is an even number of values, the median is the average of the two middle values.

In [44]:
X = [4, 5, 2, 3, 2, 1]

In [45]:
# sort X
X = sorted(X)

In [46]:
X

[1, 2, 2, 3, 4, 5]

In [47]:
if len(X)%2 == 0:
    print((X[int(len(X)/2)] + X[int(len(X)/2) + 1])/2)
else:
    print(X[int(len(X)/2)])

3.5


 let's check with extream high and low values, what happends to median ?

`X = {2, 3, 1, 4, 6, 5}`

In [52]:
def median(X):
    # sort X
    X = sorted(X)
    # find median
    if len(X)%2 == 0:
        print((X[int(len(X)/2)] + X[int(len(X)/2) + 1])/2)
    else:
        print(X[int(len(X)/2)])

In [53]:
x = [2, 3, 1, 4, 6, 5]
median(x)

4.5


In [54]:
# added -50
x = [-50, 2, 3, 1, 4, 6, 5]

median(x)

3


- median changed from `4.5` to `3`

In [55]:
# added 50
x = [2, 3, 1, 4, 6, 5, 50]

median(x)

4


- median changed from `4.5` to `4`

### **Properties**

- **Robust to Outliers:** The median is not affected by extremely high or low values, making it a robust measure of central tendency compared to the mean.
- **Indicates Distribution:** It provides a good indication of the center of a dataset, especially when data is skewed or contains outliers.

### **3. Mode**

The mode is a measure of central tendency that represents the value that appears most frequently in a dataset. Unlike the mean and median, which focus on the center or average value, the mode identifies the most common value in the dataset.

#### **Characteristics of the Mode**

1. **Most Frequent Value:** The mode is the value that occurs with the highest frequency.
2. **Multiple Modes:**
    - **Unimodal:** A dataset with only one mode.
    - **Bimodal:** A dataset with exactly two modes.
    - **Multimodal:** A dataset with more than two modes.
    - **No Mode:** If no value repeats, the dataset is said to have no mode.

In [61]:
from scipy import stats

In [62]:
#unimode;
x = [4, 1, 2, 2, 3, 4, 4]

stats.mode(x)

ModeResult(mode=4, count=3)

- 4 (appears three times) , there for 4 is mode 

In [63]:
# Bimodal
x = [1, 1, 2, 2, 3, 4]

stats.mode(x)

ModeResult(mode=1, count=2)

-  1 and 2 (both appear twice), there for 1 and 2 are modes
- but code only count 1st ones

In [64]:
# No Mode
x = [5, 7, 9, 11]

stats.mode(x)

ModeResult(mode=5, count=1)

- All values occur exactly once.
- but code only count 1st ones

#### **Applications:**

- In Exploratory data analysis & Feature engineering
- While handling missing values